File: solutions/298-binary-tree-longest-consecutive-sequence/binary-tree-longest-consecutive-sequence.go
func longestConsecutive(root *TreeNode) int {
    return dfs(root, root.Val, 0)
}

func dfs(node *TreeNode, pValue, count int) int {
    if node == nil {
        return count
    }

    if node.Val == pValue+1 {
        count++
    } else {
        count = 1
    }

    return max(count, max(dfs(node.Left, node.Val, count), dfs(node.Right, node.Val, count)))
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

File: solutions/298-binary-tree-longest-consecutive-sequence/binary-tree-longest-consecutive-sequence.cs
public class Solution {
    public int LongestConsecutive(TreeNode root) 
    {
        return Dfs(root, root.val, 0);
    }

    private int Dfs(TreeNode node, int pValue, int count)
    {
        if (node == null)
            return count;

        count = 1 + (node.val == pValue + 1 ? count : 0);
        return Math.Max(count, Math.Max(Dfs(node.left, node.val, count), Dfs(node.right, node.val, count)));
    }
}